libs/http-ext/src/lib/matchers/match-origin/match-origin.spec.ts

```diff
@@ -4,21 +4,37 @@ import { matchOrigin } from './match-origin';
 describe.each([
   /* Using a string */
   ['https://test.com', 'https://test.com', true],
+  // @todo ['https://test.com/test', 'https://test.com', true],
   ['https://test.com', 'https://angular.io', false],
+  // @todo ['https://test.com.test.com', 'https://test.com', false],
 
   /* Using an Array */
   ['https://test.com', ['https://test.com'], true],
+  // @todo ['https://test.com/test', ['https://test.com'], true],
   ['https://test.com', ['https://angular.io'], false],
 
   /* Using a RegExp */
   ['https://test.com', /[a-z]/, true],
-  ['https://test.com', /[0-9]/, false],
+  ['https://test.com', /[0-9]/, false]
+])(
+  'matchOrigin with url: %p and matcher: %p => %p',
+  (origin, matcher, expected) => {
+    it('should check origin', () => {
+      const request = createRequest({ url: origin });
+      expect(matchOrigin(matcher)({ request })).toBe(expected);
+    });
+  }
+);
 
-  /* Using a Predicate */
-  ['https://test.com', (origin: string) => origin.startsWith('https://'), true],
-  ['http://test.com', (origin: string) => origin.startsWith('https://'), false]
-])('matchOrigin(%p, %p) => %p', (origin, matcher, expected) => {
-  it('should check origin', () => {
-    expect(matchOrigin(matcher, createRequest({ url: origin }))).toBe(expected);
-  });
-});
+describe.each([['https://test.com', true], ['http://test.com', false]])(
```

Comment:
I split the tests because jest would just log that the matcher is a function and we can't know which function it is.

Reply:
Looks better :+1: